wsapi/wsapiV2.go

```diff
@@ -722,40 +720,55 @@ func HandleV2RawData(state interfaces.IState, params interface{}) (interface{},
 		return nil, NewInvalidHashError()
 	}
 
+	dbase := state.GetDB()
 	var block interfaces.BinaryMarshallable
-	var b []byte
-
-	if block, _ = state.FetchECTransactionByHash(h); block != nil {
-		b, _ = block.MarshalBinary()
-	} else if block, _ = state.FetchFactoidTransactionByHash(h); block != nil {
-		b, _ = block.MarshalBinary()
-	} else if block, _ = state.FetchEntryByHash(h); block != nil {
-		b, _ = block.MarshalBinary()
-	}
-
-	if b == nil {
-		dbase := state.GetDB()
-
-		// try to find the block data in db and return the first one found
+	switch hashkey.Type {
+	case HashTypeFCTTx:
+		block, _ = state.FetchFactoidTransactionByHash(h)
+	case HashTypeECTx:
+		block, _ = state.FetchECTransactionByHash(h)
+	case HashTypeEntry:
+		block, _ = dbase.FetchEntry(h)
+	case HashTypeEBlock:
+		block, _ = dbase.FetchEBlock(h)
+	case HashTypeECBlock:
+		block, _ = dbase.FetchECBlock(h)
+	case HashTypeFBlock:
+		block, _ = dbase.FetchFBlock(h)
+	case HashTypeABlock:
+		block, _ = dbase.FetchABlock(h)
+	case HashTypeDBlock:
+		block, _ = dbase.FetchDBlock(h)
+	default:
```

Comment:
Technically, changing the order in which blocks are searched is a change in functionality though with SHA256 I don't think the likelihood of a conflict existing in the first place is even there. This change should be fine unless I overlooked something obvious.